### PR TITLE
chore(DTFS2-6994): remove moment from portal-api facility-dates directory

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
Moment js is deprecated, it should be replaced. This PR only covers validation - see #2706, #2797 and #2691 for the rest

## Resolution :heavy_check_mark:
* Replace moment() with vanilla Dates where possible and date-fns where necessary

## Note ⚠️ 
* Need to remove `formattedTimestamp` function and package dependency on `moment` once all PRs are merged

### CODE REVIEW

File: portal-api/src/v1/facility-dates/requested-cover-start-date.js [hunk 1]
```diff
@@ -1,30 +1,19 @@
...
-  const {
-    requestedCoverStartDateDay,
-    requestedCoverStartDateMonth,
-    requestedCoverStartDateYear,
-  } = getRequestedCoverStartDateValues(facility);
-
-  const hasRequestedCoverStartDate = (hasValue(requestedCoverStartDateDay)
-    && hasValue(requestedCoverStartDateMonth)
-    && hasValue(requestedCoverStartDateYear));
+  const hasRequestedCoverStartDate =
```
Comment (AlexBramhill) [inline portal-api/src/v1/facility-dates/requested-cover-start-date.js:15]: we can just return this and no need for lines 18 to 22. If you want to keep it a named variable for readability, im happy, and we can just return that. (I know this is legacy code!)

File: portal-api/src/v1/helpers/date.test.ts [hunk 2]
```diff
@@ -79,45 +79,46 @@ describe('getDateFromDayMonthYearStrings', () => {
...
     });
 
     it('should use the current year if invalid year', () => {
-      const result = getDateFromDayMonthYearStringsReplicatingMoment(mockDay, mockMonth, '####');
+      const result = getStartOfDateFromDayMonthYearStringsReplicatingMoment(mockDay, mockMonth, '####');
 
       expect(format(result, 'dd')).toEqual(mockDay);
       expect(format(result, 'MM')).toEqual(mockMonth);
       expect(format(result, 'yyyy')).toEqual(format(new Date(), 'yyyy'));
+      expect(format(result, 'yyyy')).toEqual(format(new Date(), 'yyyy'));
```
Comment (AlexBramhill) [inline portal-api/src/v1/helpers/date.test.ts:121]: dupe

File: portal-api/src/v1/controllers/loan-issue-facility.controller.js [hunk 3]
```diff
@@ -60,7 +60,11 @@ exports.updateLoanIssueFacility = async (req, res) => {
       }
 
       if (hasAllIssuedDateValues(modifiedLoan)) {
-        modifiedLoan.issuedDate = createTimestampFromSubmittedValues(req.body, 'issuedDate');
+        modifiedLoan.issuedDate = getStartOfDateFromDayMonthYearStrings(
```
[the remark this answers is not in the document]
Comment (MarRobSoftwire) [reply portal-api/src/v1/controllers/loan-issue-facility.controller.js:63]: I've tried to not replicate moment where it's not caused any problems - the date _shouldn't_ be parsed if the `requestedCoverStart-day` is invalid but some tests seem to rely on that happening

File: portal-api/src/v1/helpers/date.test.ts (hunk 2 above)
Comment (MarRobSoftwire) [reply portal-api/src/v1/helpers/date.test.ts:121]: oops!

**Review by AlexBramhill — APPROVED** [review]
lgtm

File: portal-api/src/v1/facility-dates/requested-cover-start-date.js [hunk 4]
```diff
@@ -1,59 +1,50 @@
...
-    return true;
-  }
-
-  return false;
+  return dateHasAllValues(requestedCoverStartDateDay, requestedCoverStartDateMonth, requestedCoverStartDateYear);
 };
 
 exports.hasAllRequestedCoverStartDateValues = hasAllRequestedCoverStartDateValues;
 
+// TODO: DTFS2-7024 remove this dependence on moment behaviour
```
Comment (MarRobSoftwire) [inline portal-api/src/v1/facility-dates/requested-cover-start-date.js:20]: Note to remove this

File: package-lock.json
Comment (MarRobSoftwire) [inline package-lock.json:1]: The individual package-lock.json files are redundant and should be deleted. They're no longer updated when running `npm i`
See [here ](https://docs.npmjs.com/cli/v7/using-npm/workspaces) underneath "Below is a post npm install example, given that same previous example structure of files and folders" 
```
+-- node_modules
|  `-- workspace-a -> ../workspace-a
+-- package-lock.json
+-- package.json
`-- workspace-a
   `-- package.json
```
there's no package-lock within `workspace-a`

**Review by AlexBramhill — APPROVED** [review]

**Pull request merged**